frontend/javascripts/libs/toast.tsx

```diff
@@ -142,23 +142,37 @@ const Toast = {
       });
     }
 
-    this.notificationAPI[type](toastConfig);
-
     // Make sure that toasts don't just disappear while the user has WK in a background tab (e.g. while uploading large dataset).
     // Most browsers pause requestAnimationFrame() if the current tab is not active, but Firefox does not seem to do that.
-    if (useManualTimeout && this.pendingTimeouts[key] == null) {
+    if (useManualTimeout) {
+      // In case a toast with the same key is already open, close it first.
+      this.closePendingToastsEarlyMap[key]?.();
+      let cancelledTimeout = false;
       const timeoutToastManually = async () => {
         const splitTimeout = timeout / 2;
         await animationFrame(); // ensure tab is active
         await sleep(splitTimeout);
         await animationFrame();
+        if (cancelledTimeout) {
+          // If the toast has been closed early, don't close it again.
+          return;
+        }
```

Comment:
I think this should be moved below the second sleep. Otherwise, an old close call could close a new toast if the old one was closed between the first and second sleep :thinking:

Reply:
Oh you are right 😓